### PR TITLE
Resolves #990: Check prerequisites if required enforcer Maven version is empty

### LINKED ISSUE #990: Bug: prerequisites are not detected
When doing `versions:display-plugin-updates` on this module https://github.com/maveniverse/mima/tree/75a06a1132103506614b33a2cba2229923e86260/demo/library-maven-plugin

I get this message:
```
[INFO] --------< eu.maveniverse.maven.mima.demo:library-maven-plugin >---------
[INFO] Building eu.maveniverse.maven.mima.demo:library-maven-plugin 1.0.0-SNAPSHOT [5/5]
[INFO]   from library-maven-plugin/pom.xml
[INFO] ----------------------------[ maven-plugin ]----------------------------
[INFO] 
[INFO] --- versions:2.16.0:display-plugin-updates (default-cli) @ library-maven-plugin ---
[INFO] 
[INFO] All plugins with a version specified are using the latest versions.
[INFO] 
[INFO] All plugins have a version specified.
[INFO] 
[WARNING] Project does not define minimum Maven version required for build, default is: 3.2.5
[INFO] Plugins require minimum Maven version of: 3.2.5
[INFO] 
[WARNING] Project (which is a Maven plugin) does not define required minimum version of Maven.
[WARNING] Update the pom.xml to contain
[WARNING]     <prerequisites>
[WARNING]       <maven><!-- minimum version of Maven that the plugin works with --></maven>
[WARNING]     </prerequisites>
[WARNING] To build this plugin you need at least Maven 3.2.5
[WARNING] A Maven Enforcer rule can be used to enforce this if you have not already set one up
[WARNING] See https://maven.apache.org/enforcer/enforcer-rules/requireMavenVersion.html
```

But https://github.com/maveniverse/mima/blob/75a06a1132103506614b33a2cba2229923e86260/demo/library-maven-plugin/pom.xml#L46-L48

While true, I have no enforcer to enforce build time Maven version.

Still, why does plugin claim I have no prerequisites, while I do?

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Check prerequisites if enforcer check for Maven version is not present.

~~Prerequisites are checked after enforcer -- disputable, but at least this keeps the behavior backwards compatible.~~

The plugin will choose the version which is the highest between: default maven version, prerequisites, enforcer.

@slawekjaranowski please check

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
Actually, I think it should take the greater of the two. I'll redo the PR.

Also, I don't really think that they're should be a default maven version. Currently it's 3.2.5, which is the minimum version required by the versions plugin, but the POM being analysed is executed without the versions plugin being involved and thus it's minimum version shouldn't be taken into account.

File: versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/MinimalMavenBuildVersionFinder.java [hunk 1]
```diff
@@ -1,16 +1,18 @@
 package org.codehaus.mojo.versions;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.*;
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/MinimalMavenBuildVersionFinder.java:3]: 😄

Comment (ajarmoniuk) [reply versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/MinimalMavenBuildVersionFinder.java:3]: That' spotless apply :)

Comment (ajarmoniuk) [reply versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/MinimalMavenBuildVersionFinder.java]: Corrected.

**Pull request merged**